src/compiler/crystal/tools/doc/markdown/parser.cr

```diff
@@ -115,11 +117,25 @@ class Crystal::Doc::Markdown::Parser
       .strip                     # Strip leading/trailing whitespace
       .gsub(/[\s_-]+/, '-')      # Replace `_` and leftover whitespace with `-`
 
-    @renderer.begin_header level, anchor
+    if level < @last_anchor_level
+      # Go one beyond last_anchor_level to also remove the last actual anchor.
+      until (level - 1) == @last_anchor_level
```

Comment:
This would have been much cleaner if we had a mutating `downto` method...

Reply:
There's no way to mutate an int

Reply:
Ah, fair point.

I was imagining being able to do like

```cr
@last_anchor_level = 3

@last_anchor_level.downto! 0 do
  # ...
end

@last_anchor_level # => 0
```

But yea, I guess it's a pipe dream.

Reply:
It can be done with a macro :-D